web/auth/interceptors/access_control.go

```diff
@@ -0,0 +1,134 @@
+package interceptors
+
+import (
+	"context"
+	"strings"
+
+	pb "github.com/autograde/quickfeed/ag"
+	"github.com/autograde/quickfeed/web/auth"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+// "user" role implies that user attempts to access information about himself.
```

Comment:
We should avoid string comparison for the roles; it will be faster and less error prone to use something like this:
```go
type (
	role  int
	roles []role
)

const (
	// user role implies that user attempts to access information about himself.
	user role = iota
	// group role implies that the user is a course student + a member of the given group.
	group
	// student role implies that the user is enrolled in the course with any role.
	student
	// teacher: user enrolled in the course with teacher status.
	teacher
	// owner is the creator of a manual review.
	owner
	// admin is the ...
	admin
)

var access = map[string]roles{
	"UpdateUser":           {user, admin},
	"GetEnrollmentsByUser": {user, admin},
}
```

Reply:
```go
					roles, ok := access[method]
					if ok {
						for _, role := range roles {
							switch role {
							case user:
							case group:
							case teacher:
							case admin:
							}
						}
					}
```